src/mlpack/tests/main_tests/linear_regression_test.cpp

```diff
@@ -18,49 +18,268 @@ static const std::string testName = "LinearRegression";
 
 using namespace mlpack;
 
-// Utility function to set a parameter and mark it as passed, using copy
-// semantics.
-template<typename T>
-void SetInputParam(const std::string& name, const T& value)
-{
-  CLI::GetParam<T>(name) = value;
-  CLI::SetPassed(name);
-}
-
-// Utility function to set a parameter and mark it as passed, using move
-// semantics.
+// Utility function to set a parameter and mark it as passed,
+// using copy semantics for lvalues and move semantics for rvalues
 template<typename T>
 void SetInputParam(const std::string& name, T&& value)
 {
-  CLI::GetParam<T>(name) = std::move(value);
+  CLI::GetParam<typename std::remove_reference<T>::type>(name)
+    = std::forward<T>(value);
   CLI::SetPassed(name);
 }
 
-struct LinearRegressionTestFixture
+struct LRTestFixture
 {
  public:
-  LinearRegressionTestFixture()
+  LRTestFixture()
   {
     // Cache in the options for this program.
     CLI::RestoreSettings(testName);
   }
 
-  ~LinearRegressionTestFixture()
+  ~LRTestFixture()
   {
     // Clear the settings.
     CLI::ClearSettings();
   }
 };
 
-BOOST_FIXTURE_TEST_SUITE(LinearRegressionMainTest, LinearRegressionTestFixture);
+void ResetSettings()
+{
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+}
+
+BOOST_FIXTURE_TEST_SUITE(LinearRegressionMainTest, LRTestFixture);
+
+/**
+ * Training a model with different regularization parameter
+ * and ensuring that predictions are different
+ */
+BOOST_AUTO_TEST_CASE(LRDifferentLambdas)
+{
+  // required minimal difference between solutions
+  const double delta = 0.1;
+
+  arma::mat trainX = { 1.0, 2.0, 3.0 };
+  arma::mat testX = { 4.0 };
+  arma::rowvec trainY = { 1.0, 4.0, 9.0 };
+  SetInputParam("training", trainX);
+  SetInputParam("training_responses", trainY);
+  SetInputParam("test", testX);
+  SetInputParam("lambda", 0.1);
+
+  // first solution
+  mlpackMain();
+  const double testY1 = CLI::GetParam<arma::rowvec>("output_predictions")(0);
+
+  ResetSettings();
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("training_responses", std::move(trainY));
+  SetInputParam("test", std::move(testX));
+  SetInputParam("lambda", 1.0);
+
+  // second solution
+  mlpackMain();
+  const double testY2 = CLI::GetParam<arma::rowvec>("output_predictions")(0);
+
+  // second solution has stronger regularization,
+  // so the predicted value should be smaller
+  BOOST_REQUIRE(testY1 - delta > testY2);
```

Comment:
You can use `BOOST_REQUIRE_GT` here, which will provide slightly nicer output in the event of a failing test.